### PR TITLE
Restoring ArcGIS Cache Gridset (Tiling Schemes) support to resolve 696

### PR DESCRIPTION
**Pull request by gwcFan42**
This update restores ArcGIS Cache Gridset (Tiling Schemes) support in resolution of a related bug from 2018 (issue [696](https://github.com/GeoWebCache/geowebcache/issues/696)).

This commit:

- Reverts a breaking line change from a prior commit [here](https://github.com/GeoWebCache/geowebcache/commit/7eb3d371b0ba64dafe31ce9361385e326c36fb6a) (see ArcGISCacheLayer.java lines 212/219 in diff)
- Adds ArcGISCacheLayerTest.java which confirms an ArcGISCacheLayer can be successfully initialized.

Suggested Testing:

- Rebuild GWC
- Configuring ArcGIS cache using GWC related steps here:
  - https://www.geowebcache.org/docs/latest/configuration/layers/arcgistilingschemes.html
  - Ex. can use ESRI sample cache files here: https://github.com/Esri/raster-tiles-compactcache
- Confirm can reload GWC configurations with ArcGIS cache configured  again without occurrence of bug (issue 696) error:
  - "java.lang.IllegalStateException: No ArcGISCacheGridsetConfiguration could be found"
- Other tests: 
  - Can try ArcGIS cache configuration steps above without any fixes here to replicate original issue error above for comparison.
  - Can try reverting fix line in ArcGISCacheLayer.java (only) but leaving ArcGISCacheLayerTest.java in place.
    - Confirm new ArcGISCacheLayerTest.java fails with prior line.

### CODE REVIEW

**Comment by gwcFan42** [comment]
I saw that the pull request failed some of the workflows:
- Linux GitHub CI - QA
- Linux GitHub CI - openjdk11
- Downstream integration build

I've committed a fix for the "QA" workflow failure which was specific to the updated class in this pull request.  I needed to remove a no longer used method (from the original breaking commit from 2018) to satisfy the checks.

The build command that failed the "openjdk11" test appears to be an unrelated and intermittent build issue from swift storage. I also see it causing the same workflow to fail with other unrelated commits like this one:
* https://github.com/GeoWebCache/geowebcache/runs/5076817686?check_suite_focus=true

The build command that had failed on the "Downstream" workflow runs successfully locally and the error also appears  unrelated to this update.

Please let me know if any additional action is required.  Thank you.

**Comment by jodygarnett** [comment]
I triggered the builds to see if your `openjdk11` test would pass if given another oppertunity.

File: geowebcache/arcgiscache/src/main/java/org/geowebcache/arcgis/layer/ArcGISCacheLayer.java [hunk 1]
```diff
@@ -244,7 +244,7 @@ protected boolean initializeInternal(GridSetBroker gridSetBroker) {
         final GridSetBuilder gsBuilder = new GridSetBuilder();
         GridSet gridSet = gsBuilder.buildGridset(layerName, info, layerBounds);
 
-        getGridsetConfiguration(gridSetBroker).addInternal(gridSet);
+        gridSetBroker.put(gridSet);
```
Comment (jodygarnett) [inline geowebcache/arcgiscache/src/main/java/org/geowebcache/arcgis/layer/ArcGISCacheLayer.java:247]: I am not that familiar with how grid sets are managed, is this adding the grid set publicly? rather than internally ...

File: geowebcache/arcgiscache/src/test/java/org/geowebcache/arcgis/layer/ArcGISCacheLayerTest.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
+package org.geowebcache.arcgis.layer;
```
Comment (jodygarnett) [inline geowebcache/arcgiscache/src/test/java/org/geowebcache/arcgis/layer/ArcGISCacheLayerTest.java:1]: Appreciate the test, can you add project header this this file please.

Comment (gwcFan42) [reply geowebcache/arcgiscache/src/test/java/org/geowebcache/arcgis/layer/ArcGISCacheLayerTest.java:15]: You got it. Just added.

File: geowebcache/arcgiscache/src/main/java/org/geowebcache/arcgis/layer/ArcGISCacheLayer.java (hunk 1 above)
Comment (gwcFan42) [reply geowebcache/arcgiscache/src/main/java/org/geowebcache/arcgis/layer/ArcGISCacheLayer.java:247]: This change simply reverts the line back to its prior version, reverting the breaking change to this line [here](https://github.com/GeoWebCache/geowebcache/commit/7eb3d371b0ba64dafe31ce9361385e326c36fb6a).

### Fuller background:

Testing with different GWC versions, we'd seen that the GWC ArcGIS cache steps in the official docs ([here](https://www.geowebcache.org/docs/latest/configuration/layers/arcgistilingschemes.html)) had stopped working after GeoWebCache 1.10.5 and continued to not work up through the current version, which we've also tested with.

A related [GWC issue](https://github.com/GeoWebCache/geowebcache/issues/696) was opened by others in 2018 reflecting the same GWC ArcGIS cache layer error we've seeing in testing versions up to this point and which our merge request hopefully can be used to finally close out.

Since ArcGIS Cache Layers were working in GWC 1.10.5, we'd used that to track down the code change that caused the issue ([here](https://github.com/GeoWebCache/geowebcache/commit/7eb3d371b0ba64dafe31ce9361385e326c36fb6a)) and reverted the breaking change with our merge request.

The intention of the original breaking change itself is unfortunately unclear with its general branch description "Working on Gridsets and ArcGIS Cache for configuration changes", sparse comments and the direct outcome being ArcGIS Cache Layers themselves no longer working. As its evident the updates were made without testing ArcGIS Cache Layers, we're hoping our new test file can help prevent this going forward.

In short, this reversion allows the grid set configuration to be found for the ArcGIS Cache Layer like in prior GWC versions instead of throwing the "No ArcGISCacheGridsetConfiguration" can be found exception.

Comment (jodygarnett) [reply geowebcache/arcgiscache/src/main/java/org/geowebcache/arcgis/layer/ArcGISCacheLayer.java:247]: Thanks for the background :)

**Comment by gwcFan42** [comment]
I've merged from main to address a merge conflict.  No actual changes to actual changes here (conflicting diff was in logger syntax of method being retired here).

@jodygarnett Can you please let me know if anything else is needed to get the pull request merged to main?

**Comment by cwgrant** [comment]
Any updates on getting this merged?

**Review by smithkm — APPROVED** [review]
